app/models/concerns/uid_validation_on_update.rb

```diff
@@ -0,0 +1,7 @@
+module UidValidationOnUpdate
+  def uid_not_changed
+    if persisted? && changed.include?("uid")
```

Comment:
isn't there a dynamic `[attr]_changed?` method for every attribute, ie, `uid_changed?`?